src/AppInstallerCLICore/Workflows/ManifestComparator.cpp

```diff
@@ -295,26 +315,35 @@ namespace AppInstaller::CLI::Workflow
                 }
             }
 
-            std::string ExplainInapplicable(const Manifest::ManifestInstaller& installer) override
-            {
-                std::string result = "InstallerType does not match required type: ";
-                result += InstallerTypeToString(installer.EffectiveInstallerType());
-                result += "Required InstallerTypes: ";
-                result += m_requirementAsString;
-                return result;
-            }
-
             bool IsFirstBetter(const Manifest::ManifestInstaller& first, const Manifest::ManifestInstaller& second) override
             {
-                // TODO: Current implementation assumes there is only a single installer type requirement. This needs to be updated
-                // once multiple installerType requirements and preferences are accepted.
-                UNREFERENCED_PARAMETER(first);
-                UNREFERENCED_PARAMETER(second);
-                return true;
+                if (m_preference.empty())
+                {
+                    return false;
+                }
+
+                bool isFirstInstallerTypePreferred =
+                    ContainsInstallerType(m_preference, first.EffectiveInstallerType()) ||
+                    ContainsInstallerType(m_preference, first.BaseInstallerType);
+
+                bool isSecondInstallerTypePreferred = 
+                    ContainsInstallerType(m_preference, second.EffectiveInstallerType()) ||
+                    ContainsInstallerType(m_preference, second.BaseInstallerType);
+
+                if (isFirstInstallerTypePreferred == isSecondInstallerTypePreferred)

```

Comment:
I'm a bit torn about whether making the preference ordered vs unordered. Making it unordered will limit the user's ability to prefer a specific installer type over another and all other preferences are ordered. Making it ordered seems too strong for installer type though. maybe a middle ground will be making it ordered and put this comparator in the last on line 765 so it's not that significant in the end (btw, putting the comparator to last should be done regardless of whether we decide it ordered or unordered)